### PR TITLE
fix: [#177270155] Improve accessibility flow on add card screen

### PR DESCRIPTION
**Pull request by ncannata-dev**
## Improvement of the accessibility flow in the add card section

Based on a report from a user who reports a bad user experience on the accessibility side, the following changes have been made as agreed with the part that deals with UX

## List of changes proposed in this pull request (all changes are referenced only when Talkback / VoiceOver is on)
- Removal of **_placeholders_** in the input fields as useless for a blind person and due to the current limitation of react-native in the area of ​​accessibility, to manage hints with placeholders
- When present, the **_description text_** below the input field will be ignored. Instead, it will be read as an additional hint when you click on in the input field
- When clicking on the **_label_** of the input field, after reading the text it will also be said "Header"
- When clicking on the **_input field_**, it will read the label of the field as the base hint
- When clicking on a **_link_**, after reading the text it will also be said "Link"
- When clicking on a **_data_ _field_**, it will read the label and it will also be read "2 digits for the month, 2 for the year"
- When clicking on **_secure code_,** it will read the label and it will also be read "3 digits (or 4 digits depending on the type of credit card)"
- When clicking on **_disabled button_**, in addition to what it reads by default, it now reads even if the button is disabled.

## Sviluppato e testato con TalkBack
It is necessary to test how everything responds on IOS

### CODE REVIEW

**Review by Undermaken — CHANGES_REQUESTED** [review]

File: ts/screens/wallet/AddCardScreen.tsx [hunk 1]
```diff
@@ -187,6 +189,24 @@ const AddCardScreen: React.FC<Props> = props => {
     title: I18n.t("global.buttons.back")
   };
 
+  const isScreenReaderEnabled = useScreenReaderEnabled();
+  const placeholderCard = isScreenReaderEnabled
```
[the remark this answers is not in the document]
Comment (ncannata-dev) [reply ts/screens/wallet/AddCardScreen.tsx]: Fixed as suggested ✅

File: ts/components/LabelledItem.tsx [hunk 2]
```diff
@@ -165,6 +178,10 @@ export class LabelledItem extends React.Component<Props, State> {
             />
           ) : (
             <Input
+              accessibilityLabel={this.props.accessibilityLabel}
+              accessibilityHint={`${this.props.accessibilityHint}, ${
```
[the remark this answers is not in the document]
Comment (ncannata-dev) [reply ts/components/LabelledItem.tsx]: Fixed as suggested ✅

File: ts/components/LabelledItem.tsx [hunk 3]
```diff
@@ -150,9 +160,12 @@ export class LabelledItem extends React.Component<Props, State> {
             ) : (
               <Image source={this.props.icon} style={this.props.iconStyle} />
             ))}
-
           {this.props.type === "masked" ? (
             <TextInputMask
+              accessibilityLabel={this.props.accessibilityLabel}
+              accessibilityHint={`${this.props.accessibilityHint}, ${
```
[the remark this answers is not in the document]
Comment (ncannata-dev) [reply ts/components/LabelledItem.tsx]: Fixed as suggested ✅

**Comment by Undermaken** [comment]
@ncannata-dev this PR is still WIP (fixing iOS accessibility), right?
If yes could you move it in draft state until it is not ready?

**Comment by ncannata-dev** [comment]
> @ncannata-dev this PR is still WIP (fixing iOS accessibility), right?
> If yes could you move it in draft state until it is not ready?
 
Sure!

**Review by Undermaken — APPROVED** [review]

**Review by Undermaken — APPROVED** [review]

**Pull request merged**